plugins/cpp_metrics/service/cxxmetrics.thrift

```diff
@@ -4,26 +4,72 @@ include "project/project.thrift"
 namespace cpp cc.service.cppmetrics
 namespace java cc.service.cppmetrics
 
-enum CppMetricsType
+enum CppUnitType
+{
+  AstNodeUnit = 1,
+  FileUnit = 2
+}
+
+enum CppAstNodeMetricsType
 {
   ParameterCount = 1,
   McCabe = 2,
   LackOfCohesion = 3,
   LackOfCohesionHS = 4
 }
 
-struct CppMetricsTypeName
+enum CppModuleMetricsType
+{
+  Placeholder = 1
+}
+
+struct CppAstNodeMetricsTypeName
+{
+  1:CppAstNodeMetricsType type,
+  2:string name
+}
+
+struct CppModuleMetricsTypeName
 {
-  1:CppMetricsType type,
+  1:CppModuleMetricsType type,
   2:string name
 }
 
 struct CppMetricsAstNode
 {
-  1:CppMetricsType type,
+  1:CppAstNodeMetricsType type,
   2:double value
 }
 
+struct CppAllMetricsAstNode
+{
+  1:common.AstNodeId id,
+  2:list<CppMetricsAstNode> metrics
+}
+
+struct CppMetricsModule
+{
+  1:CppModuleMetricsType type,
+  2:double value
+}
+
+struct CppAllMetricsModule
+{
+  1:common.FileId id,
+  2:list<CppMetricsModule> metrics
+}
```

Comment:
Same remarks like for the `CppMetricsAstNode` and the `CppAllMetricsAstNode` types.

Reply:
Fixed in e09e8ff80ab891cd8da090bdd74b5fe870ab6c1c.